test/parallel/xyk-pallet.API.fees.tripleTokens.test.ts

```diff
@@ -182,180 +134,163 @@ test.skip("xyk-pallet - Check required fee - User with TUR only, operation fails
-    mintLiquidity(
-      testUser1.keyRingPair,
-      firstCurrency,
-      secondCurrency,
-      new BN(100),
-      new BN(1000000),
-    ).catch((reason) => {
-      exception = true;
-      throw new Error(reason.data);
-    }),
-  ).rejects.toThrow(feeLockErrors.AccountBalanceFail);
-  expect(exception).toBeTruthy();
+
+  const saleAssetValue = swapValueThreshold.muln(2);
+
+  await signTx(
+    api,
+    Xyk.sellAsset(secondCurrency, firstCurrency, saleAssetValue),
+    testUser1.keyRingPair,
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
 });
 
-test("xyk-pallet - Check required fee - User with very few  MGA, very few KSM and very few TUR, operation fails", async () => {
-  await setupApi();
-  await setupUsers();
+test("GIVEN User has a very limited amount of GASP & a minimal amount of Eth AND the Tx is a swap below the “threshold” THEN we receive client error", async () => {
+  const api = getApi();
+  let clientError: any;
   await Sudo.batchAsSudoFinalized(
-    Assets.mintToken(TUR_ASSET_ID, testUser1, new BN(100000)),
-    Assets.mintToken(KSM_ASSET_ID, testUser1, new BN(100000)),
-    Assets.mintToken(MGA_ASSET_ID, testUser1, new BN(100000)),
+    Assets.mintToken(GASP_ASSET_ID, testUser1, feeLockAmount.divn(2)),
+    Assets.mintToken(ETH_ASSET_ID, testUser1, BN_THOUSAND),
   );
-  let exception = false;
-  await expect(
-    mintLiquidity(
+
+  const saleAssetValue = swapValueThreshold.divn(2);
+
+  try {
+    await signTx(
+      api,
+      Xyk.sellAsset(firstCurrency, secondCurrency, saleAssetValue),
       testUser1.keyRingPair,
-      firstCurrency,
-      secondCurrency,
-      new BN(100),
-      new BN(1000000),
-    ).catch((reason) => {
-      exception = true;
-      throw new Error(reason.data);
-    }),
-  ).rejects.toThrow(feeLockErrors.AccountBalanceFail);
-  expect(exception).toBeTruthy();
+    );
+  } catch (error) {
+    clientError = error;
+  }
+  expect(clientError.data).toContain(
+    "Invalid Transaction: Fee lock processing has failed either due to not enough funds to reserve or an unexpected error",
+  );
 });
 
-test.skip("BUG under discussion:xyk-pallet - when minting all MGX should pay fees with ksm ?- Check required fee - User with very few MGA, some KSM and very few TUR", async () => {
-  await setupApi();
-  await setupUsers();
+test("User, when paying with eth, have to pay 1/10000000 eth per GASP spent.", async () => {
+  const [testUser2] = setupUsers();
+
+  testUser2.addAsset(GASP_ASSET_ID);
+  testUser2.addAsset(ETH_ASSET_ID);
+
   await Sudo.batchAsSudoFinalized(
-    Assets.mintToken(TUR_ASSET_ID, testUser1, new BN(100000)),
-    Assets.mintToken(KSM_ASSET_ID, testUser1),
-    Assets.mintToken(MGA_ASSET_ID, testUser1, Assets.DEFAULT_AMOUNT),
-    Assets.mintToken(secondCurrency, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintToken(firstCurrency, testUser2, defaultCurrencyValue),
```

Comment:
Instead of using batch, use regular signTx, cause sudo could be paying some fees. Better to test this without him.

Reply:
Done